teps/0114-custom-tasks-beta.md

```diff
@@ -0,0 +1,341 @@
+only for tests. This could be a version of the [`Wait Custom Task`][wait] or [`CEL Custom Task`][cel]. We'd import this
+`Custom Task` into Tekton Pipelines to avoid a dependency on the Tekton Experimental. This `Custom Task` will not be
+packaged with the Tekton Pipelines release - it's only intended for testing in Tekton Pipelines.
+
+### Optional
+
+The work scoped here are nice-to-have ahead of the promotion - they are not blockers.
+
+#### Authoring Custom Tasks
+
+Improve the user experience of authoring Custom Tasks and their controllers.
+
+Implementing `Custom Tasks` and their controllers can be difficult for new users of the feature. We can improve the 
+authoring experience through:
+- providing SDK or boilerplate for getting started - this enhancement is proposed in [TEP-0071][tep-0071].
+- providing a guide for authoring `Custom Tasks` and their controllers, including security recommendations such as
+scoping permissions.
+
+#### Sharing Custom Tasks
+
+Make it easy to share and discover Custom Tasks.
+
+Discovering `Custom Tasks` is difficult because some exist in the experimental repository in Tekton and others are
+contributors' own repositories - this makes it hard to share and reuse `Custom Tasks`. The Tekton Hub is the platform
+for users to discover and share Tekton resources. We propose that the Hub supports `Custom Tasks`. 
+
+##### Custom Tasks in Tekton Hub
+
+This is an overview of a potential solution for supporting Custom Tasks in the Hub and its CLI. Further details will be 
+discussed in these issues - [tektoncd/community#667][667] and [tektoncd/cli#1623][1623] - and in its own TEP if needed. 
+
+To support Custom Tasks in the Tekton Hub, we will add `customtasks.yaml` that will be used to list `Custom Tasks` to 
+be surfaced in the Hub, in the same way that [config.yaml][config] is used for Catalogs with `Tasks` and `Pipelines`.
+
+Each entry in `customtasks.yaml` will specify a `Custom Task` surfaced in the Hub, for example:
+
+```yaml
+custom-tasks:
+  
+  - name: approval
+    org: automatiko-io
+    type: community
+    provider: github
+    url: https://github.com/automatiko-io/automatiko-approval-task
+    revision: main
+    categories: 
+      - deployment
+      - automation
+  
+  - name: task-group
+    org: openshift-pipelines
+    type: community
+    provider: github
+    url: github.com/openshift-pipelines/tekton-task-group
+    revision: main
+    categories:
+      - storage 
+      
+  - name: cel
+    org: tektoncd
+    type: official
+    provider: github
+    url: github.com/tektoncd/cel
+    revision: main
+    tags:
+      - language 
+```
+
+We will require that the listed `Custom Tasks` must have `README.md` documenting its installation and functionality, 
+which will be surfaced in the Hub. The [Hub CLI][hub-cli] will support `Custom Tasks` that are surfaced in the Hub:
+
+```shell
+$ tkn hub install custom task approval
```

Comment:
@dibyom This is a good question, at least on the hub / possible catalog side of things, what a "Custom Task" implies (is it a yaml payload with the CRDs, the controller, …) and how it can be "shipped" by the hub. From there, we can definitely code this in the cli 👼🏼

Reply:
another option is to provide the custom tasks controllers through the operator -- and the instances of using the custom tasks are in the hub

Reply:
agreed, I think this question can be out of scope for this TEP and we can decide on it when we actively work on custom task + catalog/hub integration